SpecFlowProjectMars1/StepDefinitions/LanguageSkills.cs

```diff
@@ -0,0 +1,143 @@
+﻿using NUnit.Framework;
+using OpenQA.Selenium.Chrome;
+using OpenQA.Selenium.Support.UI;
+using OpenQA.Selenium;
+using SpecFlowProjectMars1.Pages;
+using SpecFlowProjectMars1.Utilities;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using System.ComponentModel.DataAnnotations;
+using SeleniumExtras.WaitHelpers;
+
+namespace SpecFlowProjectMars1.StepDefinitions
+{
+    [Binding]
+    public class LanguageSkills:CommonDriver
+    {
+        private IWebDriver driver;
+        private WebDriverWait wait;
+        Language languageobj = new Language();
+        Skill skillobj=new Skill();
+
+        [Given(@"the user logs into ProjectMars")]
+        public void GivenTheUserLogsIntoProjectMars()
+        {
+            driver = new ChromeDriver();
+            driver.Manage().Timeouts().ImplicitWait = TimeSpan.FromSeconds(10);
+            driver.Navigate().GoToUrl("http://localhost:5000/Home");
+
+            LoginActions loginPageobj = new LoginActions();
+            loginPageobj.Login(driver,"test123456@test.com", "test123456");
+        }
+        [Scope(Tag = "language")]
+        [When(@"the user adds the language ""([^""]*)""  with level ""([^""]*)""")]
+        public void WhenTheUserAddsTheLanguageWithLevel(string languages, string level)
+        {
+            languageobj.AddNewLanguage(driver,languages,level);  
+        }
+        [Then(@"the ""([^""]*)""should be added to the list")]
+        public void ThenTheShouldBeAddedToTheList(string language)
+        {
+            if (!string.IsNullOrWhiteSpace(language))
+            {
+                bool isAdded = languageobj.VerifyLanguageAdded(driver, language);
+                Assert.IsTrue(isAdded, $"The language '{language}'  was not added to the list.");
+            }
+        }
+        [Scope(Tag = "language")]
+        [When(@"I add a language ""([^""]*)"" with level ""([^""]*)""")]
+         public void WhenIAddALanguageWithLevel(string oldLang, string oldLevel)
+         {
+             languageobj.AddNewLanguage(driver,oldLang,oldLevel);
+         }
+
+        [When(@"I update ""([^""]*)"" to ""([^""]*)""")]
+        public void WhenIUpdateTo(string oldLang, string newLang)
+        {
+            languageobj.EditLanguage(driver,oldLang,newLang);
+        }
+
+        [Then(@"""([^""]*)"" is added to list")]
+        public void ThenIsAddedToList(string newLang)
+        {
+            languageobj.VerifyLanguageUpdated(driver,newLang);
```

Comment:
Please do not pass the driver In action methods (methods which in pages doing things like create, delete etc ) instead get the pages to extend to common driver where you will create a common driver class which will initialise the web driver